packages/utils-worker/tests/test-perf.ts

```diff
@@ -0,0 +1,315 @@
+
+    for (let j = 0; j < 5; j++) {
+        for (let i = 0; i < deep; i++) {
+            result += `<div>
+            <img src="icon.png"><button>Test</button>
+            `;
+        }
+
+        for (let i = 0; i < deep; i++) {
+            result += `</div>`;
+        }
+    }
+
+    result += `</form>
+        </dialog>
+    </body>
+`;
+
+    const js = resources.filter((res) => {
+        return res.type === ResourceType.JS;
+    });
+
+    if (js.length) {
+        for (const { path } of js) {
+            result += `<script src="${path}"></script>`;
+        }
+    }
+
+    result += `
+</html>
+`;
+
+    return result;
+};
+
+const prefixOrderCSS = [generateCSS('./index.css')];
+const leadingDotClasslistJS = [generateJS('./index.js')];
+
+const disabledAxeHints: HintsConfigObject = {
+    'axe/aria': 'off',
+    'axe/color': 'off',
+    'axe/forms': 'off',
+    'axe/keyboard': 'off',
+    'axe/language': 'off',
+    'axe/name-role-value': 'off',
+    'axe/other': 'off',
+    'axe/parsing': 'off',
+    'axe/semantics': 'off',
+    'axe/sensory-and-visual-cues': 'off',
+    'axe/structure': 'off',
+    'axe/tables': 'off',
+    'axe/text-alternatives': 'off',
+    'axe/time-and-media': 'off'
+};
+
+const tests: Test[] = [{
+    expectedHints: ['axe/text-alternatives'],
+    expectedTime: 22500,
+    html: generateHTML(),
+    name: 'Axe perf test',
+    timeout: 40000
+}, {
+    expectedHints: ['axe/text-alternatives'],
+    expectedTime: 37000,
+    html: generateDeepHTML(),
+    name: 'Axe deep perf test',
+    timeout: 60000
+}, {
+    expectedHints: ['button-type'],
+    expectedTime: 1650,
+    hints: disabledAxeHints,
+    html: generateDeepHTML(),
+    name: 'Button type perf test',
+    timeout: 10000
+}, {
+    expectedHints: ['x-content-type-options'],
+    expectedTime: 145,
+    hints: disabledAxeHints,
+    html: generateHTML(),
+    name: 'x-content-type-options perf test',
+    timeout: 5000
+}, {
+    expectedHints: ['css-prefix-order'],
+    expectedTime: 600,
+    hints: disabledAxeHints,
+    html: generateDeepHTML(prefixOrderCSS),
+    name: 'CSS prefix order perf test',
+    resources: prefixOrderCSS,
+    timeout: 5000
+}, {
+    expectedHints: ['create-element-svg'],
+    expectedTime: 1250,
+    hints: disabledAxeHints,
+    html: generateDeepHTML(leadingDotClasslistJS),
+    name: 'Create SVG element perf test',
+    resources: leadingDotClasslistJS,
+    timeout: 6000
+}];
+
+const runTest = (test: Test) => {
+    return getResults({
+        userConfig: {
+            hints: test.hints || [],
+            language: 'en-us'
+        }
+    }, test, console.log);
+};
+
+const delta = 15 / 100; // 15%
+
+const removeMaxMin = (results: RunResult[]): RunResult[] => {
+    const copyResults = [...results];
+
+
+    copyResults.sort((a, b) => {
+        return a.totalTime - b.totalTime;
+    });
+
+    copyResults.splice(0, 1);
+    copyResults.splice(copyResults.length - 1, 1);
+
+    return copyResults;
```

Comment:
Nit: I think using `slice` to make a copy makes this a bit simpler.

```suggestion
    return copyResults.slice(1, -1);
```